airbyte_cdk/sources/declarative/resolvers/components_resolver.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from dataclasses import InitVar, dataclass
+from typing import Any, Dict, Iterable, List, Mapping, Optional, Type, Union
+
+from deprecated.classic import deprecated
+
+from airbyte_cdk.sources.declarative.interpolation import InterpolatedString
+from airbyte_cdk.sources.source import ExperimentalClassWarning
+
+
+@dataclass(frozen=True)
+class ComponentMappingDefinition:
+    """Defines the key-value mapping configuration for a stream component."""
```

Comment:
It's likely a me problem that I don't understand what exactly this class does without looking back at the spec, but I want us to start documenting our python code thoroughly, so the reference docs site makes more sense, and that it's easier for us to onboard new team members in the future. 

In this particular case, perhaps another couple sentences or a link to where / how this thing is used would be good. Internal links to google docs with specs are also fine.

Reply:
Added more details to the docs for this class.